src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -16,12 +16,30 @@ library GPv2Encoding {
         uint32 validTo;
         uint32 appData;
         uint256 feeAmount;
-        OrderKind kind;
+        bytes32 kind;
         bool partiallyFillable;
     }
 
-    /// @dev An enum describing an order kind, either a buy or a sell order.
-    enum OrderKind {Sell, Buy}
+    /// @dev The marker value for a sell order for computing the order struct
+    /// hash. This allows the EIP-712 compatible wallets to display a
+    /// descriptive string for the order kind (intead of 0 or 1) and is.
+    ///
+    /// This valud is pre-computed from the following expression:
```

Comment:
```suggestion
    /// descriptive string for the order kind (instead of 0 or 1).
    ///
    /// This value is pre-computed from the following expression:
```